accepted/00000-reboot-required-flag.md

```diff
@@ -0,0 +1,59 @@
+- Maintaining and updating the execution module to accommodate future OS updates or changes becomes crucial. Additionally, handling new distributions or versions might require continuous modifications to the module.
+
+### Alternatives
+
+#### Salt Beacon
+For transactional systems (i.e. SLE Micro), our current approach for detecting and showing the reboot required indication is based on a Salt Beacon that runs each 10 seconds directly in the client system and notifies the server whenever the need for reboot changes. This strategy will remain for transactional systems, but it would also be possible to extend it for other Linux distributions.
+
+The main advantages of this strategy are:
+ - Instantly detect the need for a reboot, regardless of what caused it.
+ - Have a standardized way of checking for reboot required.
+
+On the other hand, the main reason for not using this alternative is the risk of overloading the server with beacon events, especially for large installations. For example, applying a patch that makes a reboot to be required in multiple systems will prompt these client systems to transmit the beacon event in addition to package profile update, resulting in a doubling of the events that the server must process. Likewise, upon the client system reboot, the beacon will send the reboot required events just after the startup event.
+
+
+
+#### Hybrid (Salt Beacon + execution module)
+It's feasible to introduce a configuration property that determines the activation or deactivation of the reboot information beacon. Subsequently, the server would detect and display the 'reboot required' indication based on this property. If the beacon is active, the execution module's outcomes are ignored, and the flag isn't removed during the processing of startup event results. Conversely, when the beacon is inactive, we resort to employing the execution module strategy. Moreover, an automatic disabling of the beacon could be implemented if the server surpasses a certain threshold of registered minions, considering that performance issues would likely arise primarily in large installations.
+
+However, the principal drawback of this approach lies in its inherent complexity, posing challenges in its implementation, comprehension, and maintenance in the future.
+
+
+### Unresolved Questions
+- Should the strategy for identifying reboot required for SUSE distros be replaced for the sake of performance and standardization with other distros? Or is it a better idea to don’t change it considering that it is running for a long time and therefore it is stable?
+- Is checking reboot information at the times of package changes (package profile update) sufficient to cover the most relevant use cases? While this strategy is currently employed for SUSE distros, should we be concerned with addressing other potential causes for reboots?
```

Comment:
Since you are considering reworking the whole thing I would not restrict the use case of other causes